src/compiler/scala/tools/nsc/backend/jvm/GenBCode.scala

```diff
@@ -28,34 +32,29 @@ abstract class GenBCode extends SubComponent {
 
   override def newPhase(prev: Phase) = new BCodePhase(prev)
 
+
   class BCodePhase(prev: Phase) extends StdPhase(prev) {
     override def description = "Generate bytecode from ASTs using the ASM library"
 
     override val erasedTypes = true
 
-    private val globalOptsEnabled = {
-      import postProcessorFrontendAccess._
-      compilerSettings.optInlinerEnabled || compilerSettings.optClosureInvocations
-    }
-
+    private var generatedHandler:ClassHandler = _
```

Comment:
This var field doesn't fit in the backend's design. Could this be a component of `CodeGen`? If there's state that needs to be re-initialized, it would be good to fit it into the `perRunLazy` infrastructure.

Reply:
probably could be a per-run. Have you measure the overhead of per-run, or considered the option of an assigned clearable structure, that is directly accessed?

Its effectively a local variable, but because the run loops via apply to run for each CompilationUnit it needs to be shared between the 2 methods and cant be passed directly